CDTDatastoreTests/CDTSessionCookieInterceptorTests.m

```diff
@@ -324,4 +356,146 @@ - (void)testCookieInterceptorRenewsEarly
+    [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+        XCTAssert([testCookieHeaderValue isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+        XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+        return [OHHTTPStubsResponse responseWithJSONObject:@{@"ok" : @(YES)} statusCode:200 headers:@{}];
+    }];
+    
+    // get resource fails, pretend it expired by returning 403 status code and response
+    [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+        XCTAssert([testCookieHeaderValue isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+        XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+        return [OHHTTPStubsResponse responseWithJSONObject:@{@"error": error,
+                                                             @"reason": reason}
+                                                statusCode:403 headers:@{}];
+    }];
+    
+    if([error containsString:@"credentials_expired"]) {
+        // call to _session endpoint to renew cookie
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssertNil(request.allHTTPHeaderFields[@"Cookie"]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"POST"]);
+            XCTAssert([request.URL.lastPathComponent isEqualToString:@"_session"]);
+            XCTAssert([request.allHTTPHeaderFields[@"Content-Type"] isEqualToString:@"application/x-www-form-urlencoded"]);
+            NSString *httpBody = [[NSString alloc] initWithData: request.OHHTTPStubs_HTTPBody encoding:NSUTF8StringEncoding];
+            XCTAssert([httpBody isEqualToString:@"name=username403&password=password"]);
+            return [OHHTTPStubsResponse
+                    responseWithJSONObject:@{
+                                             @"ok" : @(YES),
+                                             @"name" : @"username",
+                                             @"roles" : @[ @"_admin" ]
+                                             }
+                    statusCode:200
+                    headers:@{
+                              @"Set-Cookie" : [NSString
+                                               stringWithFormat:@"%@; Version=1; Path=/; HttpOnly; Max-Age=86400",
+                                               testCookieHeaderValue2]
+                              }];
+        }];
+        
+        // get resource successfully using new cookie
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssert([testCookieHeaderValue2 isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+            return [OHHTTPStubsResponse responseWithJSONObject:@{@"ok" : @(YES)} statusCode:200 headers:@{}];
+        }];
+    } else {
+        // get resource fails, cookie was not re-newed and is empty
```

Comment:
The cookie should not be empty here, it was never expired and so doesn't need renewing. It should still be on the subsequent requests.

Reply:
Fixed in 9b2f6bb.